hack/tools/go.mod

```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/cluster-api-provider-ibmcloud/hack/tools
 
-go 1.22.0
+go 1.22.10
```

Comment:
right now we are using `1.22.9` version, we need to limit to that or bump the version for the project before this change

cc @Amulyam24 @Prajyot-Parab

Reply:
Changed version from `1.22.10` to `1.22.9` to be in line with our go version.